src/Compilers/Core/Portable/CommandLine/AnalyzerConfigSet.cs

```diff
@@ -183,8 +183,14 @@ public AnalyzerConfigOptionsResult GetOptionsForSourcePath(string sourcePath)
 
             var sectionKey = _sectionKeyPool.Allocate();
 
-            var normalizedPath = PathUtilities.NormalizeWithForwardSlash(sourcePath);
-            normalizedPath = PathUtilities.ExpandAbsolutePathWithRelativeParts(normalizedPath);
+            // Note: GetDirectoryName collapses multiple slashes into a single slash and hence
+            // makes the path consistent and comparable with AnalyzerConfig.NormalizedDirectory
+            // which normalizes the directory path similarly.
+            var normalizedDirectory = sourcePath.Length == 0 ? sourcePath : Path.GetDirectoryName(sourcePath) ?? sourcePath;
+            normalizedDirectory = PathUtilities.NormalizeWithForwardSlash(normalizedDirectory);
+            normalizedDirectory = PathUtilities.ExpandAbsolutePathWithRelativeParts(normalizedDirectory);
+
+            var normalizedPath = PathUtilities.EnsureTrailingSeparator(normalizedDirectory) + Path.GetFileName(sourcePath);
```

Comment:
The `EnsureTrailingSeparator` call is going to add an OS specific separator while I believe this code path expects to have forward slashes.

Reply:
`EnsureTrailingSeparator` looks for existing slashes in the path and adds a consistent one, so that shouldn't be a problem.